text/3478-better-spread-operator.md

```diff
@@ -0,0 +1,106 @@
+- Feature Name: better-spread-operator
+- Start Date: 2023-07-23
+- RFC PR: [rust-lang/rfcs#3478](https://github.com/rust-lang/rfcs/pull/3478)
+
+# Summary
+[summary]: #summary
+
+- Allow a value to be spread into another value of a different type
+- Allow partial field spreading
+
+# Motivation
+[motivation]: #motivation
+
+We currently have the spread operator, which allows a struct to be constructed with default values provided by other instance of the struct:
+
+```rust
+struct A {
+  foo: u8,
+  bar: u8
+}
+let a = A {foo: 1, bar: 2};
+let b = A {foo: 5, ..a};
+```
+
+But this has a caveat, we can't spread another value of a different type, even if it has exactly the same fields with the same types.
```

Comment:
The reason why I bring up the semantics of functional update syntax is because this distinction is exactly why you can't do this. It makes absolutely no sense to say "fill in `A` with the fields from `B`" because that distinction is completely dependent on the layouts of both of those types. The fact that their fields are the same is totally coincidental, and also explicitly not something the user should know unless they control both the types, which is not usually the case in Rust because fields have privacy.